test/node/double_tests.js

```diff
@@ -35,4 +36,74 @@ describe('Double', function () {
       });
     }
   });
+
+  describe('specialValues', () => {
+    function twiceSerialized(value) {
```

Comment:
The function is called "twiceSerialized", but that's not really descriptive of what's actually done, since we're actually serializing once, and then deserializing it. What's done here is is generally referred to as "roundTripping", or you can be more explicit by saying `serializeThenDeserialize(value)` though even that is missing the last part of it, which is the extraction of the actual js value at the end.

It also wouldn't hurt to mention why we are only interested in promoteValues = true and what happens if promoteValues = false. I'm sure something can be done in a combination of the test case `description`, code comments, and the actual `it('should <behavior we are interested in verifying>')` to help make clear what we are testing and why in this set of tests.

Reply:
Sounds good, I've added more descripted titles for the test cases and function